chain/src/note_source.rs

```diff
@@ -6,17 +6,25 @@ use serde::{Deserialize, Serialize};
 #[serde(try_from = "pb::NoteSource", into = "pb::NoteSource")]
 pub enum NoteSource {
     Transaction { id: [u8; 32] },
+    Unknown,
     Genesis,
     FundingStreamReward { epoch_index: u64 },
     ProposalDepositRefund { proposal_id: u64 },
 }
 
+impl Default for NoteSource {
+    fn default() -> Self {
+        Self::Unknown
+    }
+}
```

Comment:
This lets component implementations fill in a Default note source if one is not set, for instance when we're doing a unit test.